### ISSUE TITLE
[Tests] Remove tests for simulate

### ISSUE THREAD

**Issue by daniel-martic-sociomantic** [sociomantic-tsunami/nessie-ui#534]
## Currently

Some tests are testing if the driver method that uses `simulate` was called with corresponding action (e.g. `click`):

```
driver.click();
expect( simulate ).toBeCalledWith( 'click' );
```
## Problem

Since this is only related to tests, it doesn't make sense to test if method `simulate` was called with "correct" argument/action, because this is Enzyme specific thing. Once if we move from using Enzyme to something else, this tests will be obsolete. Besides, it can't be called with anything other than argument that is specified in driver (e.g. `click(){ this.wrapper.simulate('click') }` can only be called with `'click'` as argument)

## Wanted

Rather than checking if `simulate` was assigned the "correct" action, we should be checking that driver method that's using that `simulate` triggers the corresponding prop - which is what we're trying to achieve with `simulate` in the first place.
Basically:
```
driver.click();
expect( onClick ).toBeCalledTimes( 1 );
```
note: `onClick` here is the prop of the component
